src/fenic/_backends/cloud/execution.py

```
@@ -164,16 +169,67 @@ def save_as_table(
     ) -> QueryMetrics:
         """Execute the logical plan and save the result as a table."""
         logger.debug(f"Saving plan {logical_plan} as table: {table_name}")
+
+        if isinstance(logical_plan, TableSink):
```

if (somehow) `logical_plan` were not a `TableSink` we could just fail fast right? it might make sense to instead do something like 

```python
if not isinstance(logical_plan, TableSink):
     raise InternalError("Non TableSink LogicalPlan passed to `save_as_table`")

if not logical_plan.location: # field will definitely be present since we have now checked that logical_plan is a TableSink
                raise ValidationError(
                    f"Cannot save to table '{table_name}' - location is required. "
                    f"Provide a location.")
```

so that later when you call `logical_plan.location` it is guaranteed to be there.

---

No longer need the table location to be part of the Sink node.